src/natcap/invest/utils.py

```diff
@@ -58,28 +113,6 @@ def capture_gdal_logging():
     Returns:
         ``None``
     """
-    osgeo_logger = logging.getLogger('osgeo')
-
-    def _log_gdal_errors(err_level, err_no, err_msg):
-        """Log error messages to osgeo.
-
-        All error messages are logged with reasonable ``logging`` levels based
-        on the GDAL error level.
-
-        Args:
-            err_level (int): The GDAL error level (e.g. ``gdal.CE_Failure``)
-            err_no (int): The GDAL error number.  For a full listing of error
-                codes, see: http://www.gdal.org/cpl__error_8h.html
-            err_msg (string): The error string.
-
-        Returns:
-            ``None``
-        """
-        osgeo_logger.log(
-            level=GDAL_ERROR_LEVELS[err_level],
-            msg='[errno {err}] {msg}'.format(
-                err=err_no, msg=err_msg.replace('\n', ' ')))
-
     gdal.PushErrorHandler(_log_gdal_errors)
```

Comment:
also curious if you've ever tried `gdal.UseExceptions()`?

Reply:
Yea, I think you nailed it James. Back when we looked at this I believe it only listened to CE_Error or CE_Fatal. This is great but when I come across some errors or maybe they were warnings that should have been errors, the handler was not raising any exception and just hiding the error. That seemed pretty dangerous. 

Here's a fun stackexchange I posted from 2012! https://gis.stackexchange.com/questions/30445/is-there-a-way-to-properly-have-gdal-raise-exceptions-in-python

And the ticket I filed that was closed when they moved everything to Github 2 years ago.
https://trac.osgeo.org/gdal/ticket/4800

And what Gdal has in their "gotchas" section currently:
https://gdal.org/api/python_gotchas.html#python-bindings-do-not-raise-exceptions-unless-you-explicitly-call-useexceptions

I'd be curious if they ever built out that setting anymore.

Reply:
For what it's worth, the example I used in my ticket and SE post is now seemingly handled by `gdal.UseExceptions()`.

```python
>>> gdal.UseExceptions()
>>> array = band.ReadAsArray(xoff=129500, yoff=64700, win_xsize=110, win_ysize=80)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ddenu\Workspace\NatCap\Repositories\venv-space\py38-latest\lib\site-packages\osgeo\gdal.py", line 3834, in ReadAsArray
    return gdalnumeric.BandReadAsArray(self, xoff, yoff,
  File "C:\Users\ddenu\Workspace\NatCap\Repositories\venv-space\py38-latest\lib\site-packages\osgeo\gdal_array.py", line 408, in BandReadAsArray
    _RaiseException()
  File "C:\Users\ddenu\Workspace\NatCap\Repositories\venv-space\py38-latest\lib\site-packages\osgeo\gdal_array.py", line 246, in _RaiseException
    raise RuntimeError(gdal.GetLastErrorMsg())
RuntimeError: realized_sedimentdeposition_latitude_masked.tif, band 1: Access window out of range in RasterIO().  Requested
(129500,64700) of size 110x80 on raster of 129600x64800.
```

Reply:
Made a new issue for this discussion in #638